pkg/util/util.go

```diff
@@ -46,6 +48,75 @@ func KubeconfigWriteToFile(filename string, clusterURL string, clusterTrustBundl
 	return nil
 }
 
+// GetExternalIP get the generated external IP from service
+func GetExternalIP(serviceType string) (string, error) {
+	// start mode, find local ip
+	if serviceType == "" {
+		addrs, err := net.InterfaceAddrs()
+		if err != nil {
+			return "", err
+		}
+		for _, addr := range addrs {
+			ipAddr, ok := addr.(*net.IPNet)
+			if !ok {
+				continue
+			}
+			if ipAddr.IP.IsLoopback() {
+				continue
+			}
+			if !ipAddr.IP.IsGlobalUnicast() {
+				continue
+			}
+			return ipAddr.IP.String(), nil
+		}
+	}
+
+	// deploy mode, find external ip
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return "", err
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return "", err
+	}
+
+	ns := GetComponentNamespace()
+	if serviceType == "loadbalancer" {
+		defaultServiceName := "multicluster-controlplane"
+		svc, err := clientset.CoreV1().Services(ns).Get(context.Background(), defaultServiceName, metav1.GetOptions{})
+		if err != nil {
+			return "", err
+		}
+		return svc.Status.LoadBalancer.Ingress[0].Hostname, nil
+	} else if serviceType == "nodeport" {
+		nodes, err := clientset.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+		if err != nil {
+			return "", err
+		}
+		for _, addr := range nodes.Items[0].Status.Addresses {
+			if addr.Type == "InternalIP" {
+				return addr.Address, nil
+			}
+		}
+	} else if serviceType == "clusterip" {
+		// get ocp route
+		routeClient, err := routev1.NewForConfig(config)
+		if err != nil {
+			return "", err
+		}
+
+		routeName := "multicluster-controlplane"
+		route, err := routeClient.RouteV1().Routes(ns).Get(context.Background(), routeName, metav1.GetOptions{})
```

Comment:
I think we may use a dynimaic client to do this, and there also need a todo for other cases